<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitReportReporter -->
<testsuite hostname="DESKTOP-B0D0J2Q" failures="0" tests="13" name="TestNGDAO.TestSanPhamDAO" time="0.638" errors="2" timestamp="2024-04-07T00:13:24 ICT" skipped="0">
  <testcase classname="TestNGDAO.TestSanPhamDAO" name="testDeleteWithNullModel" time="0.041"/>
  <system-out/>
  <testcase classname="TestNGDAO.TestSanPhamDAO" name="testInsertWithEmptyModel" time="0.046">
    <error message="com.microsoft.sqlserver.jdbc.SQLServerException: Cannot insert the value NULL into column &#039;MaLoai&#039;, table &#039;QLShop.dbo.SanPham&#039;; column does not allow nulls. INSERT fails." type="java.lang.RuntimeException">
      <![CDATA[java.lang.RuntimeException: com.microsoft.sqlserver.jdbc.SQLServerException: Cannot insert the value NULL into column 'MaLoai', table 'QLShop.dbo.SanPham'; column does not allow nulls. INSERT fails.
at com.fsm.utils.jdbcHelper.Update(jdbcHelper.java:51)
at ShopQuanLyAoQuan.dao.SanPhamDAO.insert(SanPhamDAO.java:26)
at TestNGDAO.TestSanPhamDAO.testInsertWithEmptyModel(TestSanPhamDAO.java:44)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
Caused by: com.microsoft.sqlserver.jdbc.SQLServerException: Cannot insert the value NULL into column 'MaLoai', table 'QLShop.dbo.SanPham'; column does not allow nulls. INSERT fails.
at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDatabaseError(SQLServerException.java:261)
at com.microsoft.sqlserver.jdbc.SQLServerStatement.getNextResult(SQLServerStatement.java:1752)
at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.doExecutePreparedStatement(SQLServerPreparedStatement.java:675)
at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement$PrepStmtExecCmd.doExecute(SQLServerPreparedStatement.java:594)
at com.microsoft.sqlserver.jdbc.TDSCommand.execute(IOBuffer.java:7739)
at com.microsoft.sqlserver.jdbc.SQLServerConnection.executeCommand(SQLServerConnection.java:4384)
at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeCommand(SQLServerStatement.java:293)
at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeStatement(SQLServerStatement.java:263)
at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.executeUpdate(SQLServerPreparedStatement.java:537)
at com.fsm.utils.jdbcHelper.Update(jdbcHelper.java:46)
... 28 more
]]>
    </error>
  </testcase> <!-- testInsertWithEmptyModel -->
  <system-out/>
  <testcase classname="TestNGDAO.TestSanPhamDAO" name="testSelectByTenSP" time="0.048"/>
  <system-out/>
  <testcase classname="TestNGDAO.TestSanPhamDAO" name="testSelectAll" time="0.044"/>
  <system-out/>
  <testcase classname="TestNGDAO.TestSanPhamDAO" name="testTimKiemByMaSPOrTenSP" time="0.043"/>
  <system-out/>
  <testcase classname="TestNGDAO.TestSanPhamDAO" name="testUpdateWithValidModel" time="0.052"/>
  <system-out/>
  <testcase classname="TestNGDAO.TestSanPhamDAO" name="testInsertWithNullModel" time="0.043">
    <error message="com.microsoft.sqlserver.jdbc.SQLServerException: Cannot insert the value NULL into column &#039;MaSP&#039;, table &#039;QLShop.dbo.SanPham&#039;; column does not allow nulls. INSERT fails." type="java.lang.RuntimeException">
      <![CDATA[java.lang.RuntimeException: com.microsoft.sqlserver.jdbc.SQLServerException: Cannot insert the value NULL into column 'MaSP', table 'QLShop.dbo.SanPham'; column does not allow nulls. INSERT fails.
at com.fsm.utils.jdbcHelper.Update(jdbcHelper.java:51)
at ShopQuanLyAoQuan.dao.SanPhamDAO.insert(SanPhamDAO.java:26)
at TestNGDAO.TestSanPhamDAO.testInsertWithNullModel(TestSanPhamDAO.java:36)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
Caused by: com.microsoft.sqlserver.jdbc.SQLServerException: Cannot insert the value NULL into column 'MaSP', table 'QLShop.dbo.SanPham'; column does not allow nulls. INSERT fails.
at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDatabaseError(SQLServerException.java:261)
at com.microsoft.sqlserver.jdbc.SQLServerStatement.getNextResult(SQLServerStatement.java:1752)
at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.doExecutePreparedStatement(SQLServerPreparedStatement.java:675)
at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement$PrepStmtExecCmd.doExecute(SQLServerPreparedStatement.java:594)
at com.microsoft.sqlserver.jdbc.TDSCommand.execute(IOBuffer.java:7739)
at com.microsoft.sqlserver.jdbc.SQLServerConnection.executeCommand(SQLServerConnection.java:4384)
at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeCommand(SQLServerStatement.java:293)
at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeStatement(SQLServerStatement.java:263)
at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.executeUpdate(SQLServerPreparedStatement.java:537)
at com.fsm.utils.jdbcHelper.Update(jdbcHelper.java:46)
... 28 more
]]>
    </error>
  </testcase> <!-- testInsertWithNullModel -->
  <system-out/>
  <testcase classname="TestNGDAO.TestSanPhamDAO" name="testInsertWithValidModel" time="0.073"/>
  <system-out/>
  <testcase classname="TestNGDAO.TestSanPhamDAO" name="testDeleteWithValidModel" time="0.043"/>
  <system-out/>
  <testcase classname="TestNGDAO.TestSanPhamDAO" name="testUpdateWithNullModel" time="0.054"/>
  <system-out/>
  <testcase classname="TestNGDAO.TestSanPhamDAO" name="testUpdateWithEmptyModel" time="0.051"/>
  <system-out/>
  <testcase classname="TestNGDAO.TestSanPhamDAO" name="testSelectById" time="0.049"/>
  <system-out/>
  <testcase classname="TestNGDAO.TestSanPhamDAO" name="testDeleteWithEmptyModel" time="0.051"/>
  <system-out/>
</testsuite> <!-- TestNGDAO.TestSanPhamDAO -->
